### PR TITLE
Smallfiles - use the new ES method to retrieve data from internal ES

### PR DESCRIPTION
**Pull request by Avilir**
This PR is implementing the new way of using internal elastic-search server within the cluster and retrieving data from it to the main elastic-search server.

without this fix the test will not work properly, since we saw that the old method did not work all the time.

it also implement new Library for the Performance & Scale tests to prevent code duplication.

### CODE REVIEW

File: tests/e2e/performance/test_small_file_workload.py [hunk 1]
```diff
@@ -500,43 +424,33 @@ def test_smallfile_workload(
...
+                full_results.aggregate_host_results()
+                test_status = full_results.aggregate_samples_results()
+                full_results.es_write()
+
+                # Creating full link to the results on the ES server
+                log.info(f"The Result can be found at : {full_results.results_link()}")
+            else:
+                test_status = True
+
+        assert test_status, "Test Failed !"
```
Comment (keesturam) [inline tests/e2e/performance/test_small_file_workload.py:456]: is this indentation correct?

File: ocs_ci/ocs/perftests.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+import time
+import logging
+
+from elasticsearch import Elasticsearch
+
+from ocs_ci.framework.testlib import BaseTest
+
+from ocs_ci.ocs import defaults, constants
+from ocs_ci.framework import config
+from ocs_ci.framework.pytest_customization.marks import *  # noqa: F403
```
[the remark this answers is not in the document]
Comment (Avilir) [reply ocs_ci/ocs/perftests.py:11]: this is to ignore the Fake8 Error F403,
From Flake docs :

F403 | ‘from module import *’ used; unable to detect undefined names

File: ocs_ci/ocs/perftests.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            self.crd_data["spec"]["elasticsearch"] = {
+                "server": ip,
+                "port": port,
+                "url": f"http://{ip}:{port}",
+            }
+            log.info(f"Going to use the ES : {self.crd_data['spec']['elasticsearch']}")
+
+    def set_storageclass(self, interface):
+        """
+        Setting the benchmark CDR storageclass
```
[the remark this answers is not in the document]
Comment (Avilir) [reply ocs_ci/ocs/perftests.py:103]: will fix it

File: tests/e2e/performance/test_small_file_workload.py (hunk 1 above)
Comment (Avilir) [reply tests/e2e/performance/test_small_file_workload.py:456]: yes it is, the assert is in the end, outside the if statement which start on line 436

**Review by keesturam — APPROVED** [review]

**Review by ypersky1980 — APPROVED** [review]

**Pull request merged**